src/core/thread/discover_scanner.cpp

```diff
@@ -154,7 +154,7 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
     mScanChannel = Mac::ChannelMask::kChannelIteratorFirst;
     mState       = (mScanChannels.GetNextChannel(mScanChannel) == OT_ERROR_NONE) ? kStateScanning : kStateScanDone;
 
-    otLogInfoMle("Send Discovery Request (%s)", destination.ToString().AsCString());
+    Mle::Log("Send Discovery Request", destination);
```

Comment:
Further reduce code size ?
```suggestion
    Mle::Log("Send", "Discovery Request", destination);
```
That is to say, `Log` has two arguments: `aAction` and `aWhat`. 
We can save code size because most `aAction` would be `Send` or `Receive` so compiler will use the same strings in `.rodata` section. 
Many `aWhat` arguments  would also be equal so they can be reused.

Reply:
I noticed the same... :)... that is that the MLE message string is being repeated in different logs. My idea for this is to introduce the "MLE message type" as an `enum` (or use the existing `Mle::Header::Command` enumeration) that is then converted to a string. This would be bigger change and probably better for a future PR.

I would suggest not optimizing by using same literal strings as we would then rely on the compiler/toolchain to do this optimization (i.e. use one actual instance for all same string literal definitions). While most advanced compilers would do this, the behavior is toolchain dependent.

Reply:
Using `MLE message type` enum might not be a good fit for the problem? For example, we can't convert this code just using enum. 
```c++
Log("Send Child ID Request - short", destination);
```

But we can convert it to 
```c++
Log("Send", "Child ID Request - short", destination);
```
Furthermore, this optimization should be common for compilers. 
But I am fine to apply this optimization in a future PR.

Reply:
> Using MLE message type enum might not be a good fit for the problem

I also noticed that we have some special cases  :)....(for example for "Child Update" as well)
That's why I mentioned to possibly introduce a new "MLE message type" enum (may be a super-set of `Header::Command`) for logging purposes.

> Furthermore, this optimization should be common for compilers.

I have been surprised in past (many times) by optimizations/behaviors that I expected a toolchain would do to realize later that was not supported. I would always suggest not to rely on toolchain optimizations as much as possible and try to optimize the code itself. Specially since OT is a generic network stack firmware and can be built by vendor/hardware-specific toolchains.

> That being said, I am fine to apply this optimization in a future PR.

👍